reference/glossary.md

```diff
@@ -89,6 +89,10 @@ Application Virtual Machine.
 A [VM](#vm) that is intended for running software applications. 
 Typically a TemplateBasedVM, but may be a StandaloneVM. Never a TemplateVM.
 
+TemplateBasedAppVM
+------------------
+Any [AppVM](#appvm) that depends on a TemplateVM for its root filesystem.
```

Comment:
This is tautology. There are no AppVMs that are not based on TemplateVMs, and in R4.0 anything that is based on TemplateVM is AppVM.

Reply:
> This is tautology. There are no AppVMs that are not based on TemplateVMs,

Not according to the [current official definition](https://www.qubes-os.org/doc/glossary/#appvm): "Application Virtual Machine. A VM that is intended for running software applications. Typically a TemplateBasedVM, but may be a StandaloneVM. Never a TemplateVM."

> and in R4.0 anything that is based on TemplateVM is AppVM.

That's conflating AppVMs with TemplateBasedVMs. As the definition above shows, "AppVM" originally meant "VM for running applications" (hence the "App" part), regardless of whether it's standalone or based on a template. If this conflation was enforced in code beginning with 4.0, then that was a mistake.